### ISSUE TITLE
Crossvalidation needs to be generalised

### ISSUE THREAD

**Issue by sfluegel05** [ChEB-AI/python-chebai#11]
# Status Quo
- Inner cross-validation (i.e., between train and validation set) is possible via creating each of k iterations as files and then doing k runs manually

# Goal
- Automate inner cross-validation by combining all k iterations into a single run without creating k different data sets and starting k runs manually

# Ideas
- previous, scrapped implementation 
- pytorch fabric
